### ISSUE TITLE
Weights not saving after model tuning

### ISSUE THREAD

**Issue by IsaacTieu** [ultralytics/ultralytics#19286]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Ultralytics YOLO Component

Hyperparameter Tuning

### Bug

After calling model.tune(), my weights are not saved into the directory. My other files are correctly saved (.yaml, .pngs, .csv). I have attached my code below (testing on COCO8).

![Image](https://github.com/user-attachments/assets/cdb79511-5197-4fc0-bad8-71b028485fce)

### Environment

External Supercomputer using JupyterLab

Assume no read/write issues and assume no permission issues.

### Minimal Reproducible Example

`from ultralytics import YOLO
import matplotlib.pyplot as plt
import torch

model = YOLO("yolo11n.pt") 

device = 'cuda' if torch.cuda.is_available() else 'cpu'
model.to(device)

model.tune(
    data="test.yaml",
    epochs=30,
    iterations=200,
    optimizer="AdamW",
    plots=False,
    save=True,
    val=True,
)`

test.yaml (downloaded the dataset straight from Ultralytics so assume no formatting issues):
`train: /datasets/coco8/images/train
val: /datasets/coco8/images/val  

nc: 8

names: ['person', 'bicycle', 'car', 'motorbike', 'aeroplane', 'bus', 'train', 'truck']`

### Additional

I just did a really simple run using typical tutorial datasets. Are there any ideas based on the assumptions why the model weights aren't saving? How long does it take to save?

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @IsaacTieu, thank you for your interest in Ultralytics 🚀! We appreciate you taking the time to detail the issue. 

We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users, where you can find comprehensive guides and examples for [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage. Many of the most common questions may already be addressed there!

If this is a 🐛 Bug Report, thank you for providing details and code. Since it is crucial to resolve issues effectively, we kindly ask for a fully [minimum reproducible example (MRE)](https://docs.ultralytics.com/help/minimum-reproducible-example/) so we can thoroughly debug it. Please ensure that all code provided is complete, self-contained, and runnable.

A few recommendations to verify your issue:
- Ensure you are running the latest `ultralytics` package along with its dependencies. You can upgrade to the latest version in a [**Python>=3.8**](https://www.python.org/) environment as shown below:

```bash
pip install -U ultralytics
```

- Verify your setup is compatible with all dependencies, including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn) if running on GPU, [PyTorch>=1.8](https://pytorch.org/) preinstalled, and a proper write-access directory to save files.

## Environments

YOLO can be run in up-to-date environments, and you may explore these options as a cross-check to isolate setup-specific issues:
- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Community Engagement

Join the Ultralytics community to collaborate, learn, and share:
- For real-time chat: [Discord](https://discord.com/invite/ultralytics) 🎧
- For in-depth discussions: [Discourse](https://community.ultralytics.com/)
- For sharing knowledge: [Subreddit](https://reddit.com/r/Ultralytics)

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are passing, verifying the correct operation of all [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu.

This is an automated response to ensure you have the pointers needed to begin troubleshooting 🔍. One of our engineers will review your ticket soon and provide additional assistance. Thank you for your patience and contribution! 😊

**Comment by Y-T-G** [comment]
It's saved from my test.

```python
In [5]: !ls runs/detect/tune/weights/
best.pt  last.pt
```

**Comment by IsaacTieu** [comment]
> Did you complete the run?

Yes I completed the run. Are you able to check how much space the .pt files take up? I think I have a 100GB limit so that may possible be why. Although, the hyperparameters didn't change at all, so I am not sure how much space it would take up.

**Comment by IsaacTieu** [comment]
Still not sure what the error is, but for now I am just going to bypass it with torch.save()

**Comment by Y-T-G** [comment]
The models take the size of the original model used during training * 2.
